packages/http/src/validator/validators/body.ts

```diff
@@ -86,12 +88,27 @@ function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema,
   );
 }
 
+/**
+ * Given a schema, return an equivalent schema that does not include `allOf`.
+ */
+function withoutAllOf(s: JSONSchema): JSONSchema {
+  try {
+    return mergeAllOf(s, { ignoreAdditionalProperties: true });
```

Comment:
https://github.com/stoplightio/json-schema-merge-allof/blob/master/src/index.js#L477

```suggestion
    return mergeAllOf(s, { deep: false, ignoreAdditionalProperties: true });
```
`deep` is true by default

The bad news is that flipping it off makes some of the tests fail.

This is because `allOf` isn't at the top-level, but rather nested below

```json
{
  "type": "object",
  "required": [
    "name",
    "readOnlyProperty"
  ],
  "properties": {
    "name": {
      "type": "string"
    },
    "readOnlyProperty": {
      "allOf": [
        {
          "readOnly": true
        },
        {
          "type": "string"
        }
      ]
    }
  }
}
```

Reply:
```suggestion
    if ('allOf' in s) {
      return withoutAllOf(mergeAllOf(s, { deep: false, ignoreAdditionalProperties: true }));
    }

    if (isPlainObject(s.properties)) {
      return {
        ...s,
        properties: Object.entries(s.properties).reduce((result, [key, value]) => {
          result[key] = typeof value === 'boolean' ? value : withoutAllOf(value);
          return result;
        }, {}),
      };
    }
```

this should fix the issue tho.

Reply:
^ note that this should be prob using fp-ts

Reply:
@P0lip , what's the difference between your recursive allOf-replacement and calling `mergeAllOf()` with `deep: true`?  I know you've mentioned performance problems when using `deep: true`; are those problems caused by something other than the arbitrarily-deep recursion?

Reply:
Mine doesn't traverse as deeply.
json-schema-merge-allof, when deep mode is enabled,  traverses the _entire_ schema, from the very top to the very bottom.

Consider the following scenario

Input schema
```json
{
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "foo": {
      "type": "object",
      "properties": {
        "bar": {
          "allOf": [
            {
              "type": "array",
              "items": {
                "allOf": [
                  {
                    "type": "object"
                  }
                ]
              }
            }
          ]
        }
      }
    }
  }
}
```

Output
- deep (the existing code)
   ```json
   {
      "type": "object",
      "properties": {
        "name": {
          "type": "string"
        },
        "foo": {
          "type": "object",
          "properties": {
            "bar": {
              "type": "array",
              "items": {
                "type": "object"
              }
            }
          }
        }
      }
    }
   ```
   
 - my sample
   ```json
    {
      "type": "object",
      "properties": {
        "name": {
          "type": "string"
        },
        "foo": {
          "type": "object",
          "properties": {
            "bar": {
              "type": "array",
              "items": {
                "allOf": [
                  {
                    "type": "object"
                  }
                ]
              }
            }
          }
        }
      }
    }
    ```